rust-bot/src/main.rs

```diff
@@ -38,6 +38,22 @@ fn ping(ctx: &mut Context, msg: &Message) -> CommandResult {
     return Ok(());
 }
 
+#[instrument(skip(ctx))]
+#[command]
+fn pin(ctx: &mut Context, msg: &Message) -> CommandResult {
+    let message_id = msg.content.parse::<u64>()?;
```

Comment:
I couldn't confirm if parse works similar to JS where if there is a string of text and a number in it will only read that string of numbers. I couldn't get it to print any value out here. This might be because `msg.content` outputs `!pin <id>`.

We could do something like this instead:
```suggestion
fn pin(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
    let message_id = args.single::<u64>()?;
```
We could also make sure to handle if no input is found if we'd care for that.